### DISCUSSION TITLE
Chaining SQL Queries

### DISCUSSION THREAD

**Discussion by AhmedThahir** [duckdb/duckdb#11332] [Q&A]
Hi there!
Firstly, lovely work on this library.

## Requirement

This is how I usually work with polars: Chaining SQL queries using the `pipe` operation, which makes it easy to manage long projects and also for unit-testing. 

### `main.py`

```python
df = (
    input_data
    .pipe(filter_columns)
    .pipe(filter_rows)
    .collect()
)
```

### `utils.py`

```python
def filter_columns(df):
	return query(frame=df).execute("""
	select
            TIME_PERIOD as Date,
            OBS_VALUE as CPI
	from frame
	""")

def filter_rows(df):
	return query(frame=df).execute("""
	select *
	from frame
	where MEASURE = 'CPI_INDEX'
            and REF_AREA = 'AE'
            and FREQ = 'M'
            and CPI_DIV = 'ALL'
	""")
```

## Question
I want to know
- Is there a function in duckdb similar to `pipe()` in pandas and polars that allows to chain functions
- Is there any other way to achieve something similar in DuckDB?

**Comment by Tishj** [comment]
Perhaps <https://duckdb.org/docs/api/python/relational_api#constructing-relations> is helpful here?
There is no such thing as `pipe`, but you can create a relation and then select from that relation inside another relation

Similar to how we support scanning polars/pandas dataframes, that extends to relations (created through `connection.query(...)`, not `connection.execute(...)`)

**Reply by AhmedThahir** [reply]
Hi, just following up

**Comment by AhmedThahir** [comment]
Hiii thanks!

If you could give me an example with foo() and bar() functions, I would really appreciate it

**Reply by AhmedThahir** [reply]
Hi, any updates?
